src/odemis/gui/cont/acquisition.py

```diff
@@ -449,6 +453,433 @@ def open_acquisition_dialog(self):
+        else:
+            logging.debug("Not saving into file '%s' as there is no data", filename)
+
+        return data
+
+    @call_in_wx_main
+    def _on_streams_change(self, streams):
+        """
+        a VA callback that is called when the .streams VA is changed
+        """
+        # for every entry in the list, is it also present in the .streams?
+        for i in range(self._panel.streams_chk_list.GetCount() - 1, -1, -1):
+            item_stream = self._panel.streams_chk_list.GetClientData(i)
+            if item_stream not in streams:
+                self._panel.streams_chk_list.Delete(i)
+                if item_stream in self._acquiStreams.value:
+                    self._acquiStreams.value.remove(item_stream)
+                # unsubscribe the settings VA's of this removed stream
+                for va in self._get_settings_vas(item_stream):
+                    va.unsubscribe(self._on_settings_va)
+                self._update_acquisition_time()
+                # unsubscribe the name and wavelength VA's of this removed stream
+                for va in self._get_wavelength_vas(item_stream):
+                    va.unsubscribe(self._on_stream_wavelength)
+                item_stream.name.unsubscribe(self._on_stream_name)
+
+        # for every stream in .streams, is it already present in the list?
+        item_stream = [
+            self._panel.streams_chk_list.GetClientData(i)
+            for i in range(self._panel.streams_chk_list.GetCount())
+        ]
+        for s, i in zip(streams, range(len(streams))):
+            if s not in item_stream and not isinstance(s, StaticStream):
+                self._panel.streams_chk_list.Insert(s.name.value, i, s)
+                self._panel.streams_chk_list.Check(i)
+                self._acquiStreams.value.append(s)
+                # subscribe the name and wavelength VA's of this added stream
+                for va in self._get_wavelength_vas(s):
+                    va.subscribe(self._on_stream_wavelength, init=False)
+                s.name.subscribe(self._on_stream_name, init=False)
+                # subscribe the settings VA's of this added stream
+                for va in self._get_settings_vas(s):
+                    va.subscribe(self._on_settings_va)
+                self._update_acquisition_time()
+
+    @wxlimit_invocation(1)  # max 1/s
+    def _update_acquisition_time(self):
+        """
+        Updates the estimated time
+        required for acquisition
+        """
+        acq_time = acqmng.estimateTime(self._acquiStreams.value)
+        acq_time = math.ceil(acq_time)  # round a bit pessimistic
+        # TODO take into account the time of acquiring the zlevels
+        # as well as the time required for moving the actuator
+        txt = u"Estimated time: {}.".format(units.readable_time(acq_time, full=False))
+        self._panel.txt_cryosecom_est_time.SetLabel(txt)
+
+    @call_in_wx_main
+    def _on_stream_wavelength(self, _):
+        """
```

Comment:
Maybe you can call this when you add a new stream to the checklist, that way it would be put in its right place?

Reply:
Good point :ok_hand:

Reply:
But don't call `_on_stream_wavelength`, create a function `sort_streams` or something, and call it from here and on addition.

Reply:
Makes sense